pkg/tools/clientmap/refcount.go

```diff
@@ -130,3 +130,10 @@ func (m *RefcountMap) Delete(key string) (deleted bool) {
 	_, _, deleted = m.LoadAndDelete(key)
 	return deleted
 }
+
+// Drop is deleting value for a key for any count value
+// count == 1  -->  delete
+// count > 1   -->  delete
+func (m *RefcountMap) Drop(key string) {
```

Comment:
Is this map using by someone else than `connect`?

Reply:
I would also suggest creating a private map based on this one for the connect goals to prevent API leaks and also to optimize synchronization on client creating/deleting